lib/src/front_end/piece_factory.dart

```diff
@@ -235,6 +236,62 @@ mixin PieceFactory implements CommentWriter {
     pieces.give(FunctionPiece(returnTypePiece, parametersPiece));
   }
 
+  /// Creates a [TryPiece] for try statement.
+  void createTry(TryStatement tryStatement) {
+    var piece = TryPiece();
+
+    token(tryStatement.tryKeyword);
+    var tryHeader = pieces.split();
+    createBlock(tryStatement.body);
+    var tryBlock = pieces.split();
+    piece.add(tryHeader, tryBlock);
+
+    for (var i = 0; i < tryStatement.catchClauses.length; i++) {
+      var catchClause = tryStatement.catchClauses[i];
+      token(catchClause.onKeyword, after: space);
+      visit(catchClause.exceptionType, after: space);
+
+      token(catchClause.catchKeyword, after: space);
+      token(catchClause.leftParenthesis);
+      token(catchClause.exceptionParameter?.name);
+      token(catchClause.comma, after: space);
+      token(catchClause.stackTraceParameter?.name);
+      token(catchClause.rightParenthesis);
+      var catchClauseHeader = pieces.split();
```

Comment:
Thanks for the suggestion! I made some more changes because this ended up calling `visitCatchClauseParameter` and we needed to explicitly put spaces between the `AdjacentPiece`s (which I just stole from your own experimental branch).

Please take another look and let me know if anything looks off to you.

Reply:
This looks great! I hope it will be a little simpler with the upcoming API but with the current one, this looks spot on.